File: opamenu-api/OpaMenu.Domain/DTOs/Customer/CustomerResponseDto.cs
using System;

namespace OpaMenu.Domain.DTOs.Customer;

public class CustomerResponseDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string? Email { get; set; }
    public string? PostalCode { get; set; }
    public string? Street { get; set; }
    public string? StreetNumber { get; set; }
    public string? Neighborhood { get; set; }
    public string? City { get; set; }
    public string? State { get; set; }
    public string? Complement { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: opamenu-gestor/lib/features/pos/domain/models/customer_response_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'customer_response_dto.g.dart';

@JsonSerializable()
class CustomerResponseDto {
  final String id;
  final String name;
  final String phone;
  final String? email;
  final String? postalCode;
  final String? street;
  final String? streetNumber;
  final String? neighborhood;
  final String? city;
  final String? state;
  final String? complement;

  CustomerResponseDto({
    required this.id,
    required this.name,
    required this.phone,
    this.email,
    this.postalCode,
    this.street,
    this.streetNumber,
    this.neighborhood,
    this.city,
    this.state,
    this.complement,
  });

  factory CustomerResponseDto.fromJson(Map<String, dynamic> json) =>
      _$CustomerResponseDtoFromJson(json);

  Map<String, dynamic> toJson() => _$CustomerResponseDtoToJson(this);
}
